src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+                final int nCallsAtTruth = TrancheManager.countCallsAtTruth(dataManager.getData(), Double.NEGATIVE_INFINITY);
+                final TrancheManager.SelectionMetric metric = new TrancheManager.TruthSensitivityMetric(nCallsAtTruth);
+                final List<Tranche> tranches = TrancheManager.findTranches(dataManager.getData(), TS_TRANCHES, metric, VRAC.MODE);
+                tranchesStream.print(Tranche.tranchesString(tranches));
+
+                logger.info("Writing out recalibration table...");
+                dataManager.writeOutRecalibrationTable(recalWriter, getBestAvailableSequenceDictionary());
+                if (RSCRIPT_FILE != null) {
+                    logger.info("Writing out visualization Rscript file...");
+                    createVisualizationScript(dataManager.getRandomDataForPlotting(
+                            1000,
+                            positiveTrainingData,
+                            negativeTrainingData,
+                            dataManager.getEvaluationData()),
+                            goodModel,
+                            badModel,
+                            0.0,
+                            dataManager.getAnnotationKeys().toArray(new String[USE_ANNOTATIONS.size()]));
+                }
+
+                if (VRAC.MODE == VariantRecalibratorArgumentCollection.Mode.INDEL) {
+                    // Print out an info message to make it clear why the tranches plot is not generated
+                    logger.info("Tranches plot will not be generated since we are running in INDEL mode");
+                } else {
+                    // Execute the RScript command to plot the table of truth values
+                    RScriptExecutor executor = new RScriptExecutor();
+                    executor.addScript(new Resource(PLOT_TRANCHES_RSCRIPT, VariantRecalibrator.class));
+                    executor.addArgs(new File(TRANCHES_FILE).getAbsoluteFile(), TARGET_TITV);
+                    // Print out the command line to make it clear to the user what is being executed and how one might modify it
+                    logger.info("Executing: " + executor.getApproximateCommandLine());
+                    executor.exec();
+                }
+                return true;
+            }
+            catch (Exception e) {
+                //TODO: see https://github.com/broadgsa/gatk-protected/pull/19
+                // If we retain this loop, it needs to be restructured. Currently, any existing state (including data
+                // already written to output streams) is retained for the next attempt. The output streams should
+                // really be re-created and all state should be reset. Also, because we're catching Exception, this
+                // will retry even if we have bad input, such as poorly annotated training data that results in an
+                // IllegalArgumentException.
```

Comment:
We should include a test to make sure this retry feature works as intended in GATK4.